app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,34 @@
+class PostsController < ApplicationController
+  def new
+    @user = current_user
+    @post = @user.posts.new
+    @location = Location.new
+  end
+
+  def create
+    post = current_user.posts.new(build_params)
+    if post.save
+      redirect_to user_path(current_user)
+    else
+      render :new
+    end
+  end
+
+  private
+
+  def post_params
+    params.require(:post).permit(:title, :body, location: [:state, :country])
+  end
+
+  def build_params
```

Comment:
@iandouglas @alischlereth I am not sure how I feel about the setup here. I wanted the user to be able to enter a location from the posts form, and have the back end figure out if it already existed or create a new record. This works for now, I just wonder if there is a better way to do this?

Reply:
I would do the location lookup separate, I'm generally not a fan of doing if/else branching in variable assignments.
ie:
```ruby
all_params[:location] = Location.find_by_country_and_state(loc_data[:country], loc_data[:state])
unless all_params[:location]
  all_params[:location] = Location.new(loc_data)
end
```

Reply:
I like it. Thanks Ian!